design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,254 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**                                      | **Description**                                              |
+| --------------------------------------------- | ------------------------------------------------------------ |
+| Epic                                          | https://github.com/cyberark/conjur/issues/1892               |
+| Ruby CLI                                      | https://github.com/cyberark/conjur-cli                       |
+| Go CLI                                        | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI                                    | https://github.com/cyberark/conjur-api-python3               |
+| [FPM](https://github.com/jordansissel/fpm)    | For building packages for multiple platforms (RPM, DEB, OS X) |
+| [Debify](https://github.com/conjurinc/debify) | Build a package from a Ruby gem                              |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
+
+In this document we will discuss possible options for a containerless CLI and present a recommendation for which one we should pursue.
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. We should support `init`, `authn` (`login`, `authenticate`),  `policy (load)`, `list`, `show, ` `rotate_api_key` (`host` / `user`) and `fetch secrets` functionalities. These functionalities are must-haves.
```

Comment:
this list seems _really_ biased by the design of the current Ruby CLI

would it be more accurate to require methods to:
- configure connection to Conjur (may or may not include authenticating - i.e. we may not want to require users to explicitly run an `authenticate` step)
- load policy (which modes do we want to support? PUT / POST / PATCH?)
- list all resources that I have `read` permissions on
- rotate user and host API keys
- get and set secret values

what is `show` in this context?

Reply:
`show` is for showing an object. I have never actually used this one but I am guessing it returns its metadata in a list.

It is a really biased design I agree but because we are more familiar with the Ruby CLI using those names, gives some reference to the types of commands we are talking about. That way, we don't have to re-remember which commands we are referencing. 
I imagine that if we do decide to expand on the python cli, we won't be following that naming because of the wonky UX

Reply:
@sigalsax what is the use-case to run `conjur authn authenticate`? i have never used it.

Reply:
`conjur authn authenticate` gives you the authentication token for the user that is logged in
We use it in our curl requests see [here](https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Retrieve_Secret.htm?tocpath=Developer%7CREST%C2%A0APIs%7C_____8#Examplewithcurl) for example